locator/tablets.cc

```diff
@@ -616,9 +616,19 @@ future<bool> check_tablet_replica_shards(const tablet_metadata& tm, host_id this
 }
 
 class tablet_effective_replication_map : public effective_replication_map {
+    using replication_factor_t = uint16_t;
+    using replication_factor_list = utils::small_vector<replication_factor_t, 8>;
+
     table_id _table;
     tablet_sharder _sharder;
     mutable const tablet_map* _tmap = nullptr;
+    // The `_datacenter_map` stores the association between each data center
+    // and its index in the `replication_factor_list` in `_tablet_to_dc_replication_factor_map`.
+    absl::flat_hash_map<sstring, replication_factor_t> _datacenter_map;
```

Comment:
If it's an index, it shouldn't be replication_factor_t, but some other type (e.g. size_t). It's not replication factor which is stored.

Reply:
Yes, actually it's a replication factor index type, so I will rename it accordingly.